src/elements/toniq-slider/toniq-slider.element.ts

```diff
@@ -0,0 +1,407 @@
+                        : lowerLabelOffset
+                }px`;
+                upperLabel.style.left = `${
+                    upperLabelOffset < upperLabelMin
+                        ? upperLabelMin
+                        : upperLabelOffset > upperLabelMax
+                        ? upperLabelMax
+                        : upperLabelOffset
+                }px`;
+            }
+        }
+
+        function onSliderCreated(slider: Element) {
+            const element = slider as HTMLInputElement;
+            if (props.double && element.id === 'lowerSlider') {
+                (slider as HTMLInputElement).value = `${
+                    (props.value as ToniqSliderDoubleRangeValue).min
+                }`;
+            } else if (props.double && element.id === 'upperSlider') {
+                (slider as HTMLInputElement).value = `${
+                    (props.value as ToniqSliderDoubleRangeValue).max
+                }`;
+            } else {
+                (slider as HTMLInputElement).value = props.value.toString();
+            }
+
+            slider.addEventListener('input', () => {
+                if (props.double) {
+                    fillDoubleRangeSlider();
+                    if (element.id === 'lowerSlider') {
+                        lowerSliderController();
+                    } else {
+                        upperSliderController();
+                    }
+                } else {
+                    const value = parseInt(element.value);
+                    fillRangeSlider(value);
+                }
+            });
+        }
+
+        function lowerSliderController() {
+            const lowerSlider = host.shadowRoot?.querySelector('#lowerSlider') as HTMLInputElement;
+            const upperSlider = host.shadowRoot?.querySelector('#upperSlider') as HTMLInputElement;
+            if (parseInt(lowerSlider.value) > parseInt(upperSlider.value) - 2) {
+                upperSlider.value = `${parseInt(lowerSlider.value) + 2}`;
+
+                if (parseInt(upperSlider.value) === parseInt(upperSlider.max)) {
+                    lowerSlider.value = `${parseInt(upperSlider.max) - 2}`;
+                }
+            }
+        }
+
+        function upperSliderController() {
+            const lowerSlider = host.shadowRoot?.querySelector('#lowerSlider') as HTMLInputElement;
+            const upperSlider = host.shadowRoot?.querySelector('#upperSlider') as HTMLInputElement;
+
+            if (parseInt(upperSlider.value) < parseInt(lowerSlider.value) + 2) {
+                lowerSlider.value = `${parseInt(upperSlider.value) - 2}`;
+
+                if (parseInt(lowerSlider.value) === parseInt(lowerSlider.min)) {
+                    upperSlider.value = '2';
+                }
+            }
+        }
+
+        /**
+         * Re-maps a number from one range to another.
+         * https://www.arduino.cc/reference/en/language/functions/math/map/
+         */
+        function valueMap(
```

Comment:
This should probably be named something different. Maybe this?
```suggestion
        function mapRange(
```